library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -405,6 +407,7 @@ private void downloadOrContinueBatch(List<FileDownloadInfo> downloads) {
         for (FileDownloadInfo info : downloads) {
             if (!DownloadStatus.isCompleted(info.getStatus()) && !info.isSubmittedOrRunning()) {
                 download(info);
+                return;
```

Comment:
No, this was a bug where a tester found out a way to put two batches in the downloading status. Only one file at a time can be submitted as the file could belong to different batches (hence the bug).

Parallel download (if active) are dealt in a different way, not here.